iceoryx_binding_c/test/moduletests/test_client.cpp

```diff
@@ -0,0 +1,492 @@
+{
+    ::testing::Test::RecordProperty("TEST_ID", "a0775190-5672-479b-afe9-8e127abc1bc2");
+    ClientOptions defaultOptions;
+    prepareClientInit(defaultOptions);
+
+    iox_client_t sut = iox_client_init(&sutStorage, SERVICE, INSTANCE, EVENT, nullptr);
+    ASSERT_THAT(sut, Ne(nullptr));
+}
+
+TEST_F(iox_client_test, InitializingClientWithCustomOptionsWork)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "69b2da3d-fc4f-48cf-86e7-4e4365557391");
+    iox_client_options_t options;
+    iox_client_options_init(&options);
+    options.responseQueueCapacity = 456;
+    strncpy(options.nodeName, "hypnotoad is all you need", IOX_CONFIG_NODE_NAME_SIZE);
+    options.connectOnCreate = false;
+    options.responseQueueFullPolicy = QueueFullPolicy_BLOCK_PRODUCER;
+    options.serverTooSlowPolicy = ConsumerTooSlowPolicy_WAIT_FOR_CONSUMER;
+
+    ClientOptions cppOptions;
+    cppOptions.responseQueueCapacity = options.responseQueueCapacity;
+    cppOptions.connectOnCreate = options.connectOnCreate;
+    cppOptions.nodeName = options.nodeName;
+    cppOptions.responseQueueFullPolicy = QueueFullPolicy::BLOCK_PRODUCER;
+    cppOptions.serverTooSlowPolicy = ConsumerTooSlowPolicy::WAIT_FOR_CONSUMER;
+    prepareClientInit(cppOptions);
+
+    iox_client_t sut = iox_client_init(&sutStorage, SERVICE, INSTANCE, EVENT, &options);
+    ASSERT_THAT(sut, Ne(nullptr));
+}
+
+TEST_F(iox_client_test, DeinitReleasesClient)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "91311811-6741-4bca-839d-326c375e9b8c");
+    prepareClientInit();
+    iox_client_t sut = iox_client_init(&sutStorage, SERVICE, INSTANCE, EVENT, nullptr);
+    ASSERT_THAT(sut, Ne(nullptr));
+
+    iox_client_deinit(sut);
+    EXPECT_THAT(sutPort->m_toBeDestroyed.load(), Eq(true));
+}
+
+TEST_F(iox_client_test, LoanWithValidArgumentsWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "2f7ae32e-8a43-479b-beb7-6d174b791010");
+    prepareClientInit();
+    iox_client_t sut = iox_client_init(&sutStorage, SERVICE, INSTANCE, EVENT, nullptr);
+
+    void* payload = nullptr;
+    EXPECT_THAT(iox_client_loan_request(sut, &payload, 32), Eq(AllocationResult_SUCCESS));
+    EXPECT_TRUE(isPayloadInDataSegment(payload));
+    EXPECT_THAT(memoryManager.getMemPoolInfo(0).m_usedChunks, Eq(1U));
+}
+
+TEST_F(iox_client_test, LoanAlignedChunkWithValidArgumentsWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "ff897354-8213-4f13-88fe-530e29830d79");
+    prepareClientInit();
+    iox_client_t sut = iox_client_init(&sutStorage, SERVICE, INSTANCE, EVENT, nullptr);
+    constexpr uint64_t ALIGNMENT = 128;
+    void* payload = nullptr;
+    EXPECT_THAT(iox_client_loan_aligned_request(sut, &payload, 32, ALIGNMENT), Eq(AllocationResult_SUCCESS));
+    EXPECT_TRUE(isPayloadInDataSegment(payload));
+    EXPECT_THAT(reinterpret_cast<uint64_t>(payload) % ALIGNMENT, Eq(0U));
+    EXPECT_THAT(memoryManager.getMemPoolInfo(0).m_usedChunks, Eq(1U));
+}
+
+TEST_F(iox_client_test, LoanFailsWhenNoMoreChunksAreAvailable)
```

Comment:
Usually I would argue that this test is not necessary since it is already handled on the port level but you want to check the enum translation with this, correct?

Reply:
Correct!